.github/workflows/main.yaml

```
@@ -0,0 +1,44 @@
+name: Build and push redis package
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-on-push:
+    if: github.event_name == 'push'
+    runs-on: ubuntu-latest
+    steps:     
+
+      - uses: actions/checkout@v3
```

The latest version available is v4: https://github.com/actions/checkout?tab=readme-ov-file#usage

```suggestion
      - uses: actions/checkout@v4
```